crest/Assets/Crest/Crest/Scripts/OceanRenderer.cs

```diff
@@ -1324,17 +1324,34 @@ public bool Validate(OceanRenderer ocean, ValidatedHelper.ShowMessage showMessag
             {
                 if (ocean.CreateClipSurfaceData)
                 {
-                    showMessage(LodDataMgrClipSurface.ERROR_MATERIAL_KEYWORD_MISSING, ValidatedHelper.MessageType.Error, ocean.OceanMaterial);
+                    showMessage(LodDataMgrClipSurface.ERROR_MATERIAL_KEYWORD_MISSING, ValidatedHelper.MessageType.Error, ocean.OceanMaterial,
+                        (so) => FixSetMaterialKeywordEnabled(so, LodDataMgrClipSurface.MATERIAL_KEYWORD, true));
                 }
                 else
                 {
-                    showMessage(LodDataMgrClipSurface.ERROR_MATERIAL_KEYWORD_ON_FEATURE_OFF, ValidatedHelper.MessageType.Info, ocean.OceanMaterial);
+                    showMessage(LodDataMgrClipSurface.ERROR_MATERIAL_KEYWORD_ON_FEATURE_OFF, ValidatedHelper.MessageType.Info, ocean.OceanMaterial,
+                        (so) => FixSetMaterialKeywordEnabled(so, LodDataMgrClipSurface.MATERIAL_KEYWORD, false));
                 }
             }
 
             return isValid;
         }
 
+        internal static void FixSetMaterialKeywordEnabled(SerializedObject so, string keyword, bool enabled)
+        {
+            Material mat = so.targetObject as Material;
+            Undo.RecordObject(mat, $"Enable keyword {keyword}");
+            // Does apply change in viewport, but does not change shared material asset it seems..
+            if (enabled)
+            {
+                mat.EnableKeyword(keyword);
```

Comment:
The checkboxes are based off the float value. So for _CLIPSURFACE_ON keyword, there is also _ClipSurface float. So also changing the float is required:
```csharp
mat.SetFloat("_ClipSurface", 1f);
```

I added *MATERIAL_KEYWORD_PROPERTY* a little while ago which will help.

I don't think there is another way.

Reply:
thanks